fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/storage/Storage.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.storage;
+
+import net.fabricmc.fabric.api.lookup.v1.block.BlockApiLookupRegistry;
+import net.fabricmc.fabric.api.lookup.v1.item.ItemApiLookupRegistry;
+import net.fabricmc.fabric.api.transfer.v1.transaction.Transaction;
+import net.fabricmc.fabric.impl.transfer.TransferApiImpl;
+
+/**
+ * An object that can store resources.
+ * @param <T> The type of the stored resources.
+ */
+public interface Storage<T> {
```

Comment:
I know one intended method of using the lookup api is allowing people to implement these interfaces onto their block entities. One issue I see is that you can't implement multiple types of storages because of type erasure. Would it make sense to have a specific storage interface for each resource so this implementation strategy is still possible.

Reply:
This is indeed a drawback of this approach. I do however believe that having to create inner classes in such cases is better than duplicating the entire api, which would also compromise its genericity, in the sense that it can transfer any game resource, and not just fluids or items.